### ISSUE TITLE
x509: SAN dNSName is malformed

### ISSUE THREAD

**Issue by larseggert** [quic-go/quic-go#4511]
Running the `amplificationlimit` interop test locally on my Mac, i.e.,
```
python run.py  -s quic-go -c quic-go -t amplificationlimit
```
 fails with this in the logs:
```
client  | Using commit: eb1c16bd0eb270036ca04cde8fa034569913d6c7
server  | Using commit: eb1c16bd0eb270036ca04cde8fa034569913d6c7
server  | Running QUIC server.
server  | x509: SAN dNSName is malformed
```
Any ideas?

**Comment by marten-seemann** [comment]
Might be related to https://github.com/quic-interop/quic-interop-runner/pull/376.

**Comment by marten-seemann** [comment]
I can't reproduce this locally:
```
❯ ./venv/bin/python3 run.py -d -c quic-go -s quic-go -t amplificationlimit
Saving logs to logs_2024-05-19T12:56:46.
Running with server quic-go (martenseemann/quic-go-interop:latest) and client quic-go (martenseemann/quic-go-interop:latest)

Checking compliance of quic-go client
quic-go client compliant.
Checking compliance of quic-go server
quic-go server compliant.
quic-go already tested for compliance: {'quic-go': True}
Server: quic-go. Client: quic-go. Running test case: amplificationlimit
Generated random file: qlrpqqgymy of size: 5120
Requests: https://server4:443/qlrpqqgymy

Command: WAITFORSERVER=server:443 CERTS=/tmp/certs_d4o03vjg/ TESTCASE_SERVER=transfer TESTCASE_CLIENT=transfer WWW=/tmp/www_cbjqblhr/ DOWNLOADS=/tmp/download_5gf38gi2/ SERVER_LOGS=/tmp/logs_server_axgngsfj CLIENT_LOGS=/tmp/logs_client_v9e__gi4 SCENARIO="droplist --delay=15ms --bandwidth=10Mbps --queue=25 --drops_to_server=2,3,4,5,6,7" CLIENT=martenseemann/quic-go-interop:latest SERVER=martenseemann/quic-go-interop:latest REQUESTS="https://server4:443/qlrpqqgymy" VERSION="0x1"  docker compose --env-file empty.env up --abort-on-container-exit --timeout 1 sim client server
 Container sim  Recreate
 Container sim  Recreated
 Container client  Recreate
 Container server  Recreate
 Container server  Recreated
 Container client  Recreated
Attaching to client, server, sim
sim     | waiting 10s for server:443
server  | Setting up routes...
server  | Actual changes:
server  | tx-checksumming: off
server  | 	tx-checksum-ip-generic: off
server  | 	tx-checksum-sctp: off
server  | tcp-segmentation-offload: off
server  | 	tx-tcp-segmentation: off [requested on]
server  | 	tx-tcp-ecn-segmentation: off [requested on]
server  | 	tx-tcp-mangleid-segmentation: off [requested on]
server  | 	tx-tcp6-segmentation: off [requested on]
server  | Endpoint's IPv4 address is 193.167.100.100
server  | Endpoint's IPv6 address is fd00:cafe:cafe:100::100
server  | Using commit: e2fbf3cdcd4b803d237ef9752f1a555b022f47a4
server  | Running QUIC server.
client  | Setting up routes...
client  | Actual changes:
client  | tx-checksumming: off
client  | 	tx-checksum-ip-generic: off
client  | 	tx-checksum-sctp: off
client  | tcp-segmentation-offload: off
client  | 	tx-tcp-segmentation: off [requested on]
client  | 	tx-tcp-ecn-segmentation: off [requested on]
client  | 	tx-tcp-mangleid-segmentation: off [requested on]
client  | 	tx-tcp6-segmentation: off [requested on]
client  | Endpoint's IPv4 address is 193.167.0.100
client  | Endpoint's IPv6 address is fd00:cafe:cafe::100
client  | Using commit: e2fbf3cdcd4b803d237ef9752f1a555b022f47a4
client  | wait-for-it.sh: waiting 10 seconds for sim:57832
sim     | server:443 is available after 1.012585584s
sim     | Using scenario: droplist --delay=15ms --bandwidth=10Mbps --queue=25 --drops_to_server=2,3,4,5,6,7
client  | wait-for-it.sh: sim:57832 is available after 1 seconds
client  | Starting QUIC client...
client  | Client params:
client  | Test case: transfer
sim     | Dropping packet 2 (1280 bytes) from 193.167.0.100
sim     | Dropping packet 3 (36 bytes) from 193.167.0.100
sim     | Dropping packet 4 (36 bytes) from 193.167.0.100
sim     | Dropping packet 5 (33 bytes) from 193.167.0.100
sim     | Dropping packet 6 (33 bytes) from 193.167.0.100
sim     | Dropping packet 7 (33 bytes) from 193.167.0.100
client exited with code 0
Aborting on container exit...
 Container server  Stopping
 Container client  Stopping
 Container client  Stopped
 Container server  Stopped
 Container sim  Stopping
 Container sim  Stopped

Using the client's key log file.
Using the client's key log file.
Using selector: KqueueSelector
Check of downloaded files succeeded.
Server sent 9949 bytes in Handshake CRYPTO frames.
Received a 1280 byte Initial packet from the client. Amplification limit: 3840
Received a 1280 byte Handshake packet from the server. Total: 1280
Received a 1280 byte Handshake packet from the server. Total: 2560
Received a 1280 byte Handshake packet from the server. Total: 3840
Test: amplificationlimit took 14.426307s, status: TestResult.SUCCEEDED
Run took 0:00:16.802891
+---------+---------+
|         | quic-go |
+---------+---------+
| quic-go |   ✓(A)  |
|         |   ?()   |
|         |   ✕()   |
+---------+---------+
```

I'm going to close this issue, since it works both on my machine, and on the interop runner:
<img width="173" alt="image" src="https://github.com/quic-go/quic-go/assets/1478487/38e87348-e1c3-462f-b468-fe50e8a8da9c">
